modules/vpc-sc/variables.tf

```diff
@@ -160,8 +160,8 @@ variable "ingress_policies" {
       for k, v in var.ingress_policies :
       v.from.identity_type == null || contains([
         "IDENTITY_TYPE_UNSPECIFIED", "ANY_IDENTITY",
-        "ANY_USER", "ANY_SERVICE_ACCOUNT"
-      ], coalesce(v.from.identity_type, "-"))
+        "ANY_USER_ACCOUNT", "ANY_SERVICE_ACCOUNT", ""
+      ], v.from.identity_type)
```

Comment:
this is what the API has in this enum
![image](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/assets/66007/1f6d1c91-5c97-45c4-b43d-49c15142c148)

can you align on those values?

Reply:
If you populate values in the identities list, you must use "IDENTITY_TYPE_UNSPECIFIED" or "".

When we use "IDENTITY_TYPE_UNSPECIFIED" as the API expects, we encounter the bug. Terraform stores "IDENTITY_TYPE_UNSPECIFIED" as "" as @viliampucik explained. 

This is using provider version 5.11 with terraform 1.6.6. 

The only way for us to avoid an infinite difference was to use "" instead of "IDENTITY_TYPE_UNSPECIFIED" so that our terraform code matches the state. 

Altough the API reference doesn't have "" as a valid value, the provider code shows otherwise: https://github.com/hashicorp/terraform-provider-google/blob/main/google/services/accesscontextmanager/resource_access_context_manager_service_perimeter.go

We tested publishing a module and only changed the validation and it fixes it. 

We either need a fix in the provider or in the validation.

Reply:
VPC-SC code is very complicated. After also working with it for some time now, we understand! 

Yes, just adding in support for an empty string will support our use case. 

Thanks for the support!

Reply:
Shall we also raise an issue in the provider repo? Seems like the infinite diff when specifying `IDENTITY_TYPE_UNSPECIFIED` might be a provider bug.